### ISSUE TITLE
Documentation Improvement 

### ISSUE THREAD

**Issue by luisamarti** [ambitus/pyzkiln#5]
[/docs/index.md ](https://github.com/ambitus/pyzkiln/blob/main/docs/index.md) is where pyzkiln documents all of the package functions.
Currently, everything is documented in a bulleted list format. As the project grows, the documentation format needs to be improved for better readability. 

Change index.md documentation format to be more readable and extendable. Some improvement can include adding more anchor links, tables, and sections.

**Comment by MikeFultonDev** [comment]
The example seems to show a JES flow but the documentation seems to indicate this is for RACF. Is it both? If so, can you also show a RACF example?

**Comment by luisamarti** [comment]
@MikeFultonDev are you referring to the py_racf documentation?

**Comment by MikeFultonDev** [comment]
I was looking at the README and it shows pyracf stuff but then the example is something else. 
It might be useful if each of the components had a direct link to their corresponding docs (e.g. for pyracf, you have to go through a couple spots before getting to: https://github.com/ambitus/pyzkiln/tree/main/pyzkiln_core/pyracf and i hadn't seen it before)

**Comment by luisamarti** [comment]
Good point, that would be a good improvement for the documentation, thank you!
